calculator.c

```diff
@@ -0,0 +1,141 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+int numberStack[50];
+int nSTop = -1;
+int operatorStack[50];
+int opSTop = -1;
+
+int evaluateExp(char exp[]);
+void calculate();
+int operatorPrecedence(char operator);
+int custom_isdigit(char ch);
+int custom_isspace(char ch);
+
+int main()
+{
+    char exp[80];
+    int result = 0;
+    fgets(exp, sizeof(exp), stdin);
+    exp[strcspn(exp, "\n")] = '\0';
+    if (strlen(exp) == 0)
+    {
+        printf("Error: Invalid expression.\n");
+        return 0;
+    }
+    result = evaluateExp(exp);
+    printf("%d\n", result);
+    return 0;
+}
+
+int evaluateExp(char exp[])
+{
+    for (int i = 0; i < strlen(exp); i++)
+    {
+        char ch = exp[i];
+        if (custom_isdigit(ch))
+        {
+            int num = 0;
+            while (custom_isdigit(ch))
+            {
+                num = num * 10 + (ch - '0');
+                i++;
+                if (i < strlen(exp))
```

Comment:
when in the loop you have added the exit condition (i < strlen(exp), is it required to check it explicitely here again ? Will the code reach this line if i>= strlen(exp)

Reply:
You're right. Since the loop condition already ensures i is within bounds, the explicit check inside the loop is redundant. I will remove that extra check